schema.md

```diff
@@ -0,0 +1,136 @@
+# Sparkle Motion schema (WIP)
+
+All record types have `createdAt` (and maybe `modifiedAt`?).
+
+### User
+
+```js
+{
+  "id": ___,
+  "username": "xian",
+  "password": "...",      // salted hash of pw
+  "name": "Full Name",
+  "priority": 123.4,      // e.g. for locking out other users during a performance
+}
+```
+
+### Gadget Presets
+
+Gadgets offer the ability to save their current values as a user preset, or to load presets.
+
+```js
+{
+  "id": ___,
+  "userId": ___,
+  "name": "Pride Flag",
+  "type": "Palette",      // the type of the gadget
+  "data": { ... },        // type-dependent
+}
+```
+
+#### Color Gadget Data
+```js
+{
+  "color": 0xfedbca,
+}
+```
+
+#### Palette Gadget Data
+```js
+{
+  "type": "analogous" | "monochromatic" | "triad" | "complementary" | "compound" | "shades" | "custom",
+
+  // e.g. for type "custom":
+  "values": [0xE70000, 0xFF8C00, 0xFFEF00, 0x00811F, 0x0044FF, 0x760089],
+}
+
+```
+
+#### Slider Gadget Data
+```js
+{
+  "value": 0.7325,
+}
+```
+
+### Show Presets
```

Comment:
Perhaps tangential to this, do we have a concept of a show playlist?

Reply:
That's an awesome idea. Do you think we want just a simple list of shows with presets that you advance through, or something more fancy e.g. timing info or cue notes or whatevs?

Reply:
I think a simple list of shows that run one after the other would be a cool start. 

The fancy timing info/cue notes feature seems like a v2 kinda thing. 

Use case for playlist of shows:
- Have each party lead pick the light shows they want to have on loop throughout the course of their party
- UI/ability to create/edit/delete a playlist on the fly (because this might need to happen 5 minutes before the party starts...)

Other ideas:
- Ability to switch between playlist mode, and individual show mode? Or should we always just rely on the concept of a playlist mode?

Reply:
Maybe if you haven't touched anything for a couple minutes the UI drops you into automatic show rotation from your playlist… otherwise the next couple shows in your playlist show up at the top of the show picker?

Reply:
Added a playlist store.